diff --git a/gtsam/CMakeLists.txt b/gtsam/CMakeLists.txt
index 16dca6736..5872aa393 100644
--- a/gtsam/CMakeLists.txt
+++ b/gtsam/CMakeLists.txt
@@ -74,12 +74,12 @@ foreach(subdir ${gtsam_subdirs})
 endforeach(subdir)
 list(APPEND gtsam_srcs ${gtsam_core_headers})
 
-IF(MSVC)
-	# Add precompiled header to sources
-	include(gtsamAddPch)
-	gtsamAddPch("precompiled_header.h" "precompiled_header.cpp" "${gtsam_srcs}")
-	list(INSERT gtsam_srcs 0 "precompiled_header.cpp")
-ENDIF(MSVC)
+# IF(MSVC)
+#	# Add precompiled header to sources
+#	include(gtsamAddPch)
+#	gtsamAddPch("precompiled_header.h" "precompiled_header.cpp" "${gtsam_srcs}")
+#	list(INSERT gtsam_srcs 0 "precompiled_header.cpp")
+# ENDIF(MSVC)
 
 # Generate and install config and dllexport files
 configure_file(config.h.in config.h)
